format-table: make all widths be set properly

nosada opened the pull request:
This commit fixes https://github.com/systemd/systemd/issues/9549.

1. Remove assert() in `table_print()`:
https://github.com/systemd/systemd/blob/9eb0fc32d6cb63e71f2cfc1e7dd34c3ede4b48a3/src/basic/format-table.c#L1142

2. Check `finalize` when breaking for loop in `table_print()`: 
https://github.com/systemd/systemd/blob/9eb0fc32d6cb63e71f2cfc1e7dd34c3ede4b48a3/src/basic/format-table.c#L1137

yuwata commented on src/basic/format-table.c:
```diff
@@ -1139,7 +1139,6 @@ int table_print(Table *t, FILE *f) {
                         if (finalize) {
-                                assert(!restart);
                                 break;
                         }
```
Please remove unnecessary `{}`.

nosada replied:
Ah, sorry I forgot it.
But thanks to your comment I noticed this deletion seems not to make any sense, because checking `finalize` has done before reaching L:1142 after changing `break` condition.

I'll revert it.

nosada replied:
My confusion... Deleting `assert()` is needed to fix https://github.com/systemd/systemd/issues/9549.

yuwata commented:
The issue seems not restricted to `machinectl`. So, please update the commit message when you update this.

nosada commented:
Updated by getting into line with @yuwata's comment.

I also updated the title of this PR because original title doesn't make sense now, but I wonder this is correct for the PR policy of yours.
Should I recreate another PR? or can I keep this up?

nosada commented:
Thanks, I understood.
And sorry for being late to tell you that `machinectl` was already removed from commit message.

yuwata commented:
BTW, it is not necessary, but if possible, could you provide any reproducers in `src/test/test-format-table.c`? Then, we can easily judge what is the problem and actually this fixes the issue.

nosada commented:
Reported issue in https://github.com/systemd/systemd/issues/9549 can be reproduced like below.
At least I checked my PR in below procedure whether the issue is fixed or not.

1. Make at least 3 images visible in `machinectl list-images`
2. Do `machinectl list-images | less` or `machinectl list-images > foo`
3. If problem is fixed, commands invoked in `2.` successed. Get assertion if not fixed.

Examples:
```
$ machinectl list-images
NAME  TYPE RO  USAGE CREATED                     MODIFIED
foo   raw  no   1.0G Wed 2018-07-11 00:10:33 JST Wed 2018-07-11 00:16:00 JST
bar   raw  no 673.7M Wed 2018-07-11 00:23:06 JST Wed 2018-07-11 00:28:06 JST
baz   raw  no   2.1G Tue 2018-07-10 23:54:08 JST Wed 2018-07-11 00:04:18 JST
qux   raw  no 687.4M Tue 2018-07-10 23:45:46 JST Tue 2018-07-10 23:50:28 JST

4 images listed.

$ machinectl list-images | less
(got `Assertion '!restart' failed at ../systemd-stable/src/basic/format-table.c:1142, function table_print(). Aborting.`)

$ machinectl list-images > hoge
Assertion '!restart' failed at ../systemd-stable/src/basic/format-table.c:1142, function table_print(). Aborting.
Aborted (core dumped)

$
```

I hope this comment will help you.

yuwata commented:
I've squashed the commits by @nosada and added a reproducer in test-format-table.c.
The test fails without the first commit.
PTAL.

nosada commented:
Sorry... I misread.
You exactly wrote about 'reproducers' in `src/test/test-format-table.c`, not sequence of commands to reproduce problem (I'm ashamed to write I thought this is called as 'reproducer'). 
And I didn't use any reproducers to create this PR... I checked my fixing by typing commands written in https://github.com/systemd/systemd/pull/9550#issuecomment-404217502 manually.

Reproducer https://github.com/systemd/systemd/pull/9550/commits/b0804bbf5c30e00c29c270fa77238ef7287d0fd3 seems to work fine because:
- this fails when https://github.com/systemd/systemd/pull/9550/commits/a26db0bc5d4011de24e5930af6fd9051deeb46be not `cherry-pick`
- after `cherry-pick` https://github.com/systemd/systemd/pull/9550/commits/a26db0bc5d4011de24e5930af6fd9051deeb46be this successed

nosada commented:
@yuwata 
I would want to ask you that there are any other points in this PR to fix for https://github.com/systemd/systemd/pull/9550#pullrequestreview-135577663.

This review seems to be answered by https://github.com/systemd/systemd/pull/9550/commits/a26db0bc5d4011de24e5930af6fd9051deeb46be (squashed by you) already and contains any other of what is needed to change for it.

yuwata commented:
CI failures are not related. Let's merge this.

nosada commented:
Oh, I see, that is claimed by CI... Thank you responding and merging.
